### PR TITLE
Add test for Cassandra-11031 allow filtering on partition key

### PR DESCRIPTION
**Pull request by ifesdjeen**
dtests for #11031 (still in review)

Puts back riptano/cassandra-dtest#969 once ticket is merged

### CODE REVIEW

File: cql_tests.py [hunk 1]
```diff
@@ -284,6 +284,138 @@ def statements_test(self):
 
         assert_one(session, "SELECT COUNT(*) FROM test7 WHERE kind = 'ev1'", [0])
 
+    @since('3.8')
+    def partition_key_allow_filtering_test(self):
+        """
+        Test for CASSANDRA-11031 : filter on part of partition keys
```
Comment (ptnapoleon) [inline cql_tests.py]: Can this docstring have `@jira_ticket CASSANDRA-11031` like the others? Thanks

Comment (ifesdjeen) [inline cql_tests.py]: Thanks for noticing! Fixed.

**Comment by ptnapoleon** [comment]
Looks like this is ready to merge. Can you change the `since` annotations to 3.10? Also, can you take the change you made in user_types_test, and have that ONLY apply to 3.10+? On 2.2 and 3.0 we should still expect the previous error. Thanks!

**Comment by ifesdjeen** [comment]
Great, I'm waiting only on a couple of missing dtests from @jasonstack

**Comment by ifesdjeen** [comment]
I've added the missing tests myself. Running CI against the latest 3.0, 2.2 and trunk just to make sure everything's all right after my adjustments.

**Comment by ifesdjeen** [comment]
Test results: 

  * [2.2](http://cassci.datastax.com/job/ifesdjeen-11031-2.2-dtest)
  * [3.0](http://cassci.datastax.com/job/ifesdjeen-11031-3.0-dtest)
  * [trunk](http://cassci.datastax.com/job/ifesdjeen-11031-trunk-dtest)

**Comment by ifesdjeen** [comment]
If you could take a short final look, it'd be great. We'd close 2 tickets with that one :)

**Comment by ptnapoleon** [comment]
+1! Feel free to merge

**Comment by jasonstack** [comment]
@ifesdjeen  sorry for the late reply.. I just came back from the states.. thanks for the help,

**Pull request merged**